internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -80,7 +83,7 @@ type ClusterDeploymentsReconciler struct {
 // +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
 // +kubebuilder:rbac:groups=hive.openshift.io,resources=clusterdeployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=hive.openshift.io,resources=clusterdeployments/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups=hive.openshift.io,resources=clusterdeployments/finalizers,verbs=update
+// +kubebuilder:rbac:groups=agent-install.openshift.io,resources=clusterdeployments/deprovision,verbs=update
```

Comment:
Can you explain this change? `ClusterDeployment` doesn't exist in API group `agent-install.openshift.io`. And what's up with the `deprovision` subresource?

I'm thinking this line was correct before this change.

Reply:
I changed this as a follow-up to the proposal @dgoodwin made in https://github.com/openshift/assisted-service/pull/1606#discussion_r625018140

You are correct to mention that the API group should remain `hive.openshift.io`, so I reverted back to that.
As far as the finalizer name goes, I see `deprovision` [is mentioned in Hive code base](https://github.com/openshift/hive/blob/5c038fb051906a7c2d733e7b017690044c66c124/apis/hive/v1/clusterdeployment_types.go#L21-L23) and `finalizers` seems like the generic name (at least [according to the docs](https://book.kubebuilder.io/reference/using-finalizers.html)).
I proposed `ai-deprovision`, to make it a finalized specific to `assisted-installer`.
let me know what you think about it.

Reply:
This is not the place to put the name of a specific finalizer. It refers to a user's ability to modify the `finalizers` field on a resource's `ObjectMeta`.

Reply:
You can see the example here: https://book.kubebuilder.io/reference/using-finalizers.html